src/task/models/tasksManager.ts

```diff
@@ -32,135 +35,153 @@ export class TasksManager {
       this.logger.error({ err: error, limit: limit, msg: errMessage });
       throw error;
     }
   }
 
-  private getExportManagerInstance(domain: Domain): IExportManager {
-    let exportManagerInstance: IExportManager;
-    const unsupportedDomainErrorMsg = `unsupported domain requested: "${domain}" - currently only "${Domain.RASTER}" domain is supported`;
-    try {
-      this.logger.debug({ msg: `getting export manager instance by domain: ${domain}`, domain });
-
-      switch (domain) {
-        case Domain.RASTER:
-          exportManagerInstance = new ExportManagerRaster(this.logger);
-          break;
-        case Domain.DEM:
-          throw new BadRequestError(unsupportedDomainErrorMsg);
-        case Domain._3D:
-          throw new BadRequestError(unsupportedDomainErrorMsg);
-        default:
-          throw new BadRequestError(unsupportedDomainErrorMsg);
-      }
-      return exportManagerInstance;
-    } catch (error) {
-      const errMessage = `failed to get export manager instance by domain, ${(error as Error).message}`;
-      this.logger.error({ err: error, domain: domain, msg: errMessage });
-      throw error;
+  private async upsertTask(
+    exportManagerInstance: IExportManager,
+    req: CreateExportTaskRequest<TaskParameters>,
+    domainResponse: CreateExportTaskResponse,
+    customerName: string
+  ): Promise<ITaskEntity> {
+    const jobId = domainResponse.jobId;
+    if (!domainResponse.status) {
+      const res = await this.createNewTask(exportManagerInstance, req, domainResponse, customerName);
+      this.logger.info({
+        msg: `created new task, id: ${res.id}`,
+        jobId,
+        customerName,
+      });
+      return res;
     }
-  }
 
-  private async upsertTask(exportManagerInstance: IExportManager, req: CreateExportTaskRequest<TaskParameters>, domainResponse: CreateExportTaskResponse, customerName?: string): Promise<ITaskEntity> {
-    const jobId = domainResponse.jobId;
-    this.logger.info({ msg: `querying for similar task by job id ${jobId} and customer name: ${customerName}`, jobId: domainResponse.jobId, customerName });
+    this.logger.debug({
+      msg: `querying for similar '${domainResponse.status}' task by job id ${jobId} and customer name: ${customerName}`,
+      jobId,
+      customerName,
+    });
     const task = await this.taskRepository.getCustomerTaskByJobId(jobId, customerName);
 
-    if (task) {
-      this.logger.info({ msg: `found similar task id ${task.id} with job id ${jobId} and customer name: ${customerName}, updating task webhooks request`, jobId, customerName, webhooks: req.webhooks });
-      await this.webhooksRepository.upsertWebhooks(req.webhooks, task.id);
-      return task;
+    if (!task) {
+      const errMessage = `task with job id ${jobId} and customer name: ${customerName} was not found`;
+      this.logger.error({
+        msg: errMessage,
+        jobId,
+        customerName,
+      });
+      throw new NotFoundError(errMessage);
```

Comment:
i don't think this is the response the client should get

Reply:
then lets talk with him...

Reply:
ok